src/Microsoft.AspNet.Server.Kestrel/Filter/ConnectionFilterContext.cs

```diff
@@ -2,12 +2,14 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System.IO;
+using System.Security.Cryptography.X509Certificates;
 
 namespace Microsoft.AspNet.Server.Kestrel.Filter
 {
     public class ConnectionFilterContext
     {
         public ServerAddress Address { get; set; }
-        public Stream Connection { get; set;  }
+        public Stream Connection { get; set; }
+        public X509Certificate2 ClientCertificate { get; set; }
```

Comment:
This doesn't fit the abstraction level. Should we have a property bag for extra stuff?